### ISSUE TITLE
ScalaGen emitControlObject Mistake

### ISSUE THREAD

**Issue by mattfel1** [stanford-ppl/spatial#89]
See SingleLayerConv_RCIO for an example.  The ifThenElse node thinks that things defined inside of its scope are inputs to the kernel.  I think it has something to do with binds not looking beyond the first layer?
```
    val ins    = func.flatMap(_.nestedInputs)
    val binds  = lhs.op.map{d => d.binds ++ d.blocks.map(_.result) }.getOrElse(Set.empty).toSeq
    val inputs = (lhs.op.map{_.inputs}.getOrElse(Nil) ++ ins).filterNot(_.isMem).distinct diff binds
```

**Comment by dkoeplin** [comment]
The culprit here was the SeriesForeach and Array[Map,Reduce,etc.] methods, none of which had the op binds fields properly defined.

**Comment by dkoeplin** [comment]
Fixed in e3d38a38ac9b7885f4c3e40026119acb3672753a
